frontend/packages/console-app/src/components/cloud-shell/Terminal.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import { Terminal as XTerminal } from 'xterm';
+import { fit } from 'xterm/lib/addons/fit/fit';
+
+const terminalOptions = {
+  fontFamily: 'monospace',
+  fontSize: 16,
+  cursorBlink: false,
+  cols: 80,
+  rows: 25,
+  padding: 4,
+};
+
+type TerminalProps = {
+  onData: (data: string) => void;
+};
+
+export type ImperativeTerminalType = {
+  focus: () => void;
+  reset: () => void;
+  onDataReceived: (data) => void;
+  onConnectionClosed: (msg: string) => void;
+};
+
+const Terminal = React.forwardRef<ImperativeTerminalType, TerminalProps>(({ onData }, ref) => {
+  const terminal = React.useRef<XTerminal>();
+  const terminalRef = React.useRef<HTMLDivElement>();
+
+  const focus = () => {
+    terminal.current && terminal.current.focus();
+  };
```

Comment:
All the imperative handle functions should be defined inside the imperative handle hook callback so that they aren't needlessly declared on each render.